features/smoke_tests/handled.feature

```diff
@@ -0,0 +1,164 @@
+    And the event "device.time" is a timestamp
+    And the event "metaData.device.locationStatus" is not null
+    And the event "metaData.device.emulator" is false
+    And the event "metaData.device.networkAccess" is not null
+    And the event "metaData.device.charging" is not null
+    And the event "metaData.device.screenDensity" is not null
+    And the event "metaData.device.dpi" is not null
+    And the event "metaData.device.screenResolution" is not null
+    And the event "metaData.device.brand" is not null
+    And the event "metaData.device.batteryLevel" is not null
+
+    # User
+    And the event "user.id" is not null
+    And the payload field "events.0.user.id" equals the stored value "device_id"
+
+    # Breadcrumbs
+    And the event has a "state" breadcrumb named "Bugsnag loaded"
+    And the event has a "manual" breadcrumb named "HandledJavaSmokeScenario"
+    And the event "breadcrumbs.2.metaData.source" equals "BreadcrumbCallback"
+
+    # MetaData
+    And the event "metaData.TestData.ClientMetadata" is true
+    And the event "metaData.TestData.CallbackMetadata" is true
+
+Scenario: Notify Kotlin exception with overwritten configuration
+    When I run "HandledKotlinSmokeScenario"
+    Then I wait to receive a request
+    And the request is valid for the error reporting API version "4.0" for the "Android Bugsnag Notifier" notifier
+
+    # Exception details
+    And the payload field "events" is an array with 1 elements
+    And the exception "errorClass" equals "java.lang.RuntimeException"
+    And the exception "message" equals "HandledKotlinSmokeScenario"
+    And the exception "type" equals "android"
+    And the payload field "events.0.exceptions.0.stacktrace" is a non-empty array
+    And the event "unhandled" is false
+    And the event "severity" equals "error"
+    And the event "severityReason.type" equals "userCallbackSetSeverity"
+
+    # Overwritten App data
+    And the event "app.releaseStage" equals "HandledKotlinSmokeScenario"
+    And the event "app.type" equals "Overwritten"
+    And the event "app.version" equals "9.9.9"
+    And the event "app.versionCode" equals 999
+
+    # Overwritten User
+    And the event "user.id" equals "ABC"
+    And the event "user.email" equals "ABC@CBA.CA"
+    And the event "user.name" equals "HandledKotlinSmokeScenario"
+
+    # Breadcrumbs
+    And the event has a "manual" breadcrumb named "HandledKotlinSmokeScenario"
+    And the event "breadcrumbs.2.metaData.Source" equals "HandledKotlinSmokeScenario"
+
+    # MetaData
+    And the event "metaData.TestData.Source" equals "HandledKotlinSmokeScenario"
+    And the event "metaData.TestData.Callback" is true
+    And the event "metaData.TestData.redacted" equals "[REDACTED]"
+
+Scenario: Handled C functionality
+    When I run "CXXNotifySmokeScenario"
+    And I wait to receive a request
+
+    # Exception details
+    Then the request payload contains a completed handled native report
```

Comment:
It doesn't look like this step validates the headers etc to the same extent as `the request is valid for the error reporting API version "4.0" for the "Android Bugsnag Notifier" notifier`. It'd probably be worth using that here instead and raising a ticket to address the existing native scenarios which use this step, unless there's a reason this step is preferred.

Reply:
What was the outcome here?

Reply:
This was addressed as suggested